### PR TITLE
Cleanup nits in code

### PR DESCRIPTION
**Pull request by suquark**
## Why are these changes needed?

After using the new formatter, many strings are not formatted correctly. I feel I could not tolerate them. This PR fixes these strings.

This PR also does some trivial cleanup, like deleting unused global variables and arguments, I did this when going through the whole codebase looking for abnormal strings (some strings cannot be captured by simple regex or there are false positives).

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by suquark** [comment]
Not sure who I should assign to review this PR 😂

**Comment by suquark** [comment]
I have manually confirmed the fix file by file.

File: python/ray/_private/runtime_env/_clonevirtualenv.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def clone_virtualenv(src_dir, dst_dir):
     logger.info("fixing scripts in bin...")
     fixup_scripts(src_dir, dst_dir, version)
 
-    has_old = lambda s: any(i for i in s if _dirmatch(i, src_dir))
+    has_old = lambda s: any(i for i in s if _dirmatch(i, src_dir))  # noqa
```
Comment (jjyao) [inline python/ray/_private/runtime_env/_clonevirtualenv.py:85]: `# noqa: E731`?

**Review by jjyao — COMMENTED** [review]
This is great!!
One suggestion: since this PR touches 223 files 😂 , could we break it into multiple smaller PRs for easy review? Several ideas:
1. Separate different types of cleanups: string fix, dead code, comment fix, etc
2. Separate based on modules/libraries/folders

**Comment by suquark** [comment]
Sure! Let me break it down into a few PRs.

**Comment by bveeramani** [comment]
Hey, thanks for making this PR! Feel free to assign me to the formatting-related PRs once you've split this up.

**Comment by suquark** [comment]
Part 1 @ https://github.com/ray-project/ray/pull/23128

File: python/ray/_private/services.py [hunk 2]
```diff
@@ -896,8 +894,7 @@ def start_reaper(fate_share=None):
             pass
         else:
             logger.warning(
-                "setpgrp failed, processes may not be "
-                "cleaned up properly: {}.".format(e)
+                "setpgrp failed, processes may not be " f"cleaned up properly: {e}."
```
Comment (jjyao) [inline python/ray/_private/services.py:897]: this can just be a single f""?

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/services.py (hunk 2 above)
Comment (suquark) [reply python/ray/_private/services.py]: sure! thanks for pointing out it!

**Pull request merged**